lib/yapay_ex/request.ex

```diff
@@ -2,23 +2,36 @@ defmodule YapayEx.Request do
   @moduledoc """
   Implement functions to request Yapay API.
   """
+  alias YapayEx.Config
 
-  use Tesla
+  def child_spec do
+    {Finch,
+     name: __MODULE__,
+     pools: %{
+       get_domain() => [size: pool_size()]
+     }}
+  end
 
-  alias YapayEx.Config
+  def post(endpoint, order) do
+    :post
+    |> Finch.build(
+      "#{get_domain()}#{endpoint}",
+      [{"Content-Type", "application/json"}],
+      encode_body(order)
+    )
+    |> Finch.request(__MODULE__)
+  end
+
+  defp pool_size, do: 25
 
-  plug(
-    Tesla.Middleware.BaseUrl,
+  defp get_domain do
     System.get_env("MIX_ENV", "test")
-    |> String.to_existing_atom()
+    |> String.to_atom()
```

Comment:
So this is not ideal, as will grow up the number of atoms and consume more memory

Reply:
@estevanjantsk you could replace this logic to use [`Mix.env/0`](https://hexdocs.pm/mix/Mix.html#env/0). This way you won't need to unsafely parse a string to atom. 

```suggestion
    Config.domain_for(Mix.env())
```

Reply:
I will fix everything x). Sorry the delay.